src/poly/dcrt/matrix/mmap_matrix.rs

```diff
@@ -64,24 +64,77 @@ impl<T: MmapMatrixElem> MmapMatrix<T> {
+        let mmap = map_file(&self.file, start_offset, total_bytes);
+        let data = mmap.to_vec();
+        drop(mmap);
+        let columns: Vec<Vec<T>> = (cols.start..cols.end)
+            .into_par_iter()
+            .map(|col| {
+                let mut col_vec = Vec::with_capacity(nrows);
+                for row in 0..nrows {
+                    let offset = row * total_cols * entry_size + col * entry_size;
+                    let elem_bytes = &data[offset..offset + entry_size];
+                    col_vec.push(T::from_bytes_to_elem(&self.params, elem_bytes));
+                }
+                col_vec
+            })
+            .collect();
+
+        columns
+    }
+
+    pub fn block_entries_row(&self, rows: Range<usize>, cols: Range<usize>) -> Vec<Vec<T>> {
+        let entry_size = self.entry_size();
+        let num_cols = cols.end - cols.start;
         parallel_iter!(rows)
             .map(|i| {
                 let offset = entry_size * (i * self.ncol + cols.start);
-                let mmap = map_file(&self.file, offset, entry_size * cols.len());
-                let row_vec = mmap.to_vec();
-                let row_col_vec = row_vec
+                let mmap = map_file(&self.file, offset, entry_size * num_cols);
+                let row_bytes = mmap.to_vec();
+                let row_entries = row_bytes
                     .chunks(entry_size)
                     .map(|entry| T::from_bytes_to_elem(&self.params, entry))
                     .collect_vec();
                 drop(mmap);
-                row_col_vec
+                row_entries
             })
-            .collect::<Vec<Vec<_>>>()
+            .collect()
     }
 
-    pub fn replace_entries<F>(&mut self, rows: Range<usize>, cols: Range<usize>, f: F)
+    pub fn replace_entries_column<F>(&mut self, rows: Range<usize>, cols: Range<usize>, f: F)
+    where
+        F: Fn(Range<usize>, Range<usize>) -> Vec<Vec<T>> + Send + Sync,
+    {
+        let (row_offsets, col_offsets) = block_offsets(rows, cols);
+        parallel_iter!(col_offsets.iter().tuple_windows().collect_vec()).for_each(
+            |(cur_block_col_idx, next_block_col_idx)| {
+                parallel_iter!(row_offsets.iter().tuple_windows().collect_vec()).for_each(
+                    |(cur_block_row_idx, next_block_row_idx)| {
+                        let new_entries = f(
+                            *cur_block_row_idx..*next_block_row_idx,
+                            *cur_block_col_idx..*next_block_col_idx,
+                        );
+                        // The non-overlapping nature of blocks ensures thread safety.
+                        unsafe {
+                            self.replace_block_entries(
+                                *cur_block_row_idx..*next_block_row_idx,
+                                *cur_block_col_idx..*next_block_col_idx,
+                                new_entries,
+                            );
+                        }
```

Comment:
The first and second dimensions correspond to the column and row, respectively, right?

Reply:
Since this output format is different from the other functions, could you add some comments to this function?

Reply:
sure, it's only to meant to used in column represented format will add comments